homeassistant/components/octoprint/__init__.py

```diff
@@ -147,169 +132,92 @@ def device_discovered(service, info):
-                self.job_error_logged = False
-                self.printer_error_logged = False
-                self.available_error_logged = False
-
-            return response.json()
-
-        except requests.ConnectionError as exc_con:
-            log_string = "Failed to connect to Octoprint server. Error: %s" % exc_con
+            job = await self._octoprint.get_job_info()
+        except Exception as error:  # pylint: disable=broad-except
+            _LOGGER.error("Failed to update job data %s", error)
+            if self.data and "job" in self.data:
+                job = self.data["job"]
 
-            if not self.available_error_logged:
-                _LOGGER.error(log_string)
-                self.job_available = False
-                self.printer_available = False
-                self.available_error_logged = True
-
-            return None
-
-        except requests.HTTPError as ex_http:
-            status_code = ex_http.response.status_code
-
-            log_string = "Failed to update OctoPrint status. Error: %s" % ex_http
-            # Only log the first failure
-            if endpoint == "job":
-                log_string = f"Endpoint: job {log_string}"
-                if not self.job_error_logged:
-                    _LOGGER.error(log_string)
-                    self.job_error_logged = True
-                    self.job_available = False
-            elif endpoint == "printer":
-                if (
-                    status_code == 409
-                ):  # octoprint returns HTTP 409 when printer is not connected (and many other states)
-                    self.printer_available = False
-                else:
-                    log_string = f"Endpoint: printer {log_string}"
-                    if not self.printer_error_logged:
-                        _LOGGER.error(log_string)
-                        self.printer_error_logged = True
-                        self.printer_available = False
-
-            self.available = False
-
-            return None
-
-    def update(self, sensor_type, end_point, group, tool=None):
-        """Return the value for sensor_type from the provided endpoint."""
-        response = self.get(end_point)
-        if response is not None:
-            return get_value_from_json(response, sensor_type, group, tool)
-
-        return response
-
-
-def get_value_from_json(json_dict, sensor_type, group, tool):
-    """Return the value for sensor_type from the JSON."""
-    if group not in json_dict:
-        return None
-
-    if sensor_type in json_dict[group]:
-        if sensor_type == "target" and json_dict[sensor_type] is None:
-            return 0
-
-        return json_dict[group][sensor_type]
-
-    if tool is not None:
-        if sensor_type in json_dict[group][tool]:
-            return json_dict[group][tool][sensor_type]
+        try:
+            printer = await self._octoprint.get_printer_info()
+        except Exception as error:  # pylint: disable=broad-except
```

Comment:
I think this is a bit too broad, especially since the UpdateCoordinator will handle all kinds of cases.

For example, in the case of a connection timeout, the `last_update_success` is now no longer set.